src/main/java/gibson/task/Parser.java

```diff
@@ -0,0 +1,191 @@
+                        + task
+                        + "\nNow you have " + taskList.size() + " task(s) in the list.";
+            } catch (IllegalArgumentException e) {
+                return e.getMessage();
+            }
+            // DEADLINES
+        } else if (Pattern.matches("(deadline .+ /by .+)|(deadline .+ /by( )?)", input)) {
+            String taskString = Parser.substringAfterToken(input, "deadline");
+            String[] stringArray = Parser.substringBeforeAfterToken(taskString, "/by");
+            try {
+                Deadline deadline = new Deadline(stringArray[0], stringArray[1]);
+                taskList.add(deadline);
+                storage.save(taskList.toString());
+                return "Got it. I've added this task:\n"
+                        + deadline
+                        + "\nNow you have " + taskList.size() + " task(s) in the list.";
+            } catch (IllegalArgumentException e) {
+                return e.getMessage();
+            }
+            // EVENTS
+        } else if (Pattern.matches("(event .+ /at .+)|(event .+ /at( )?)", input)) {
+            String taskString = Parser.substringAfterToken(input, "event");
+            String[] stringArray = Parser.substringBeforeAfterToken(taskString, "/at");
+            try {
+                Event event = new Event(stringArray[0], stringArray[1]);
+                taskList.add(event);
+                storage.save(taskList.toString());
+                return "Got it. I've added this task:\n"
+                        + event
+                        + "\nNow you have " + taskList.size() + " task(s) in the list.";
+            } catch (IllegalArgumentException e) {
+                return e.getMessage();
+            }
+            // REMOVE
+        } else if (Pattern.matches("delete [0-9]+", input)) {
+            int number = Parser.getTrailingInt(input);
+            int index = number - 1;
+            try {
+                Task t = taskList.remove(index);
+                storage.save(taskList.toString());
+                return "Noted. I've removed this task:\n"
+                        + t.toString()
+                        + "\nNow you have " + taskList.size() + " task(s) in the list.";
+            } catch (IndexOutOfBoundsException e) {
+                return "There is not task numbered as " + number + ".";
+            }
+            // FIND
+        } else if (Pattern.matches("(find .+)|(find( )?)", input)) {
+            String searchQuery = Parser.substringAfterToken(input, "find");
+            TaskList result = taskList.find(searchQuery);
+            if (!result.isEmpty()) {
+                return "Here are your results:\n" + result.toString();
+            } else {
+                return "Unable to find task with \"" + searchQuery + "\".";
+            }
+            // NOT RECOGNIZED
+        } else {
+            return "I'm sorry. I do not know what that means.";
+        }
+    }
```

Comment:
processInput method seems too long for me. 

As per 2103 Code Quality guide:

> Be wary when a method is longer than the computer screen, and take corrective action when it goes beyond 30 LOC (lines of code). The bigger the haystack, the harder it is to find a needle. ([source](https://nus-cs2103-ay2223s1.github.io/website/se-book-adapted/chapters/codeQuality.html#guideline-maximise-readability:~:text=Be%20wary%20when%20a%20method%20is%20longer%20than%20the%20computer%20screen%2C%20and%20take%20corrective%20action%20when%20it%20goes%20beyond%2030%20LOC%20(lines%20of%20code).%20The%20bigger%20the%20haystack%2C%20the%20harder%20it%20is%20to%20find%20a%20needle.))

Perhaps 1 thing that you can do is to put try catch inside each case clause to a separate method or class.